resources/assets/js/jsvalidation.js

```diff
@@ -7,96 +7,167 @@
-                }.bind(this), 500);
-
-                return true;
-            },
-
-
-            check: function( element ) {
-                element = this.validationTargetFor( this.clean( element ) );
-
-                var rules = $( element ).rules(),
-                    rulesCount = $.map( rules, function( n, i ) {
-                        return i;
-                    }).length,
-                    dependencyMismatch = false,
-                    val = this.elementValue( element ),
-                    result, method, rule;
-
-                for ( method in rules ) {
-                    rule = { method: method, parameters: rules[ method ] };
-                    try {
-                        var methodToCall=this.normalizeMethod(method);
-                        result = $.validator.methods[ methodToCall ].call( this, val, element, rule.parameters );
-
-                        // if a method indicates that the field is optional and therefore valid,
-                        // don't mark it as valid when there are no other rules
-                        if ( result === "dependency-mismatch" && rulesCount === 1 ) {
-                            dependencyMismatch = true;
-                            continue;
-                        }
-                        dependencyMismatch = false;
-
-                        if ( result === "pending" ) {
-                            this.toHide = this.toHide.not( this.errorsFor( element ) );
-                            return;
-                        }
-
-                        if ( !result ) {
-                            this.formatAndAdd( element, rule );
-                            return false;
-                        }
-                    } catch ( e ) {
-                        if ( this.settings.debug && window.console ) {
-                            console.log( "Exception occurred when checking element " + element.id + ", check the '" + rule.method + "' method.", e );
-                        }
-                        throw e;
-                    }
+    setupValidations: function () {
+
+        /**
+         * Create JQueryValidation check to validate Laravel rules
+         */
+
+        $.validator.addMethod("laravelValidation", function (value, element, params) {
+            var validator = this;
+            var validated = true;
+
+            // put Implicit rules in front
+            var rules=[];
+            $.each(params, function (i, param) {
+                if (param[3] || laravelValidation.implicitRules.indexOf(param[0])!== -1) {
+                    rules.unshift(param);
+                } else {
+                    rules.push(param);
+                }
+            });
+
+            $.each(rules, function (i, param) {
+                var implicit = param[3] || laravelValidation.implicitRules.indexOf(param[0])!== -1;
+                var rule = param[0],
+                    arguments = param[1],
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['arguments' is defined but never used.](https://www.codacy.com/app/albert-moreno-t/laravel-jsvalidation/file/842620832/issues/source?bid=2225045&fileBranchId=2805859#l48)